app/models/customer.rb

```diff
@@ -23,7 +23,17 @@ def last_name=(l_name)
   end
 
   def payment_allocations
-    payment_allocation_sets&.last&.payment_allocations || []
+    payment_allocation_sets&.first&.payment_allocations || []
+  end
+
+  def payment_allocations_for_all_organizations
+    p = payment_allocations
+    o_donated_to = p.map(&:organization_id)
```

Comment:
Do a '.pluck" call here. Or a 'joins().select"

Reply:
I need the actual payment allocations as well. Since I want to show all organizations at this time and I only store payment allocations for anything > 0% I need to load all other organizations. Is there a better way to do this?

Reply:
Yeah you could denormalize it, this is a pretty pricey thing to do.  Pre-optimization avoision and all but this will be a PITA In the future.